roles/tm-deepsec-aws-setup/tasks/main.yml

```diff
@@ -0,0 +1,86 @@
+---
+# tasks file for tm_silent_setup
+
+- name: Ensure bash, OpenSSl, and libssl are the latest versions
+  yum: name={{ item }} update_cache=true state=latest
+  with_items: "{{ pre_package }}"
+  tags: packages
+
+- name: Install psycopg2 python package for postgres
+  pip:
+    name: "{{ item }}"
+  with_items: "{{ pip_package }}"
+  #become_user: ec2-user
+
+- name: Change psycopg2 dir permission to 0755
+  file:
+    path: "{{ item }}"
+    mode: u=rwX,g=rX,o=rX
+    recurse: yes
+  with_items: "{{ dir_permission }}"
+
+- name: Remove the postgresql package
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items: "{{ uninstall_postgres }}"
+
+- name: Remove pgsql directory
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items: "{{ remove_dir }}"
+
+- name: Install postgres 11 libs rpm from a remote file
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items: "{{ postgres_rpm }}"
+
+- name: Create postgres data cluster
+  command: /usr/pgsql-11/bin/postgresql-11-setup initdb
+
+- name: Start postgres service
+  service:
+    name: postgresql-11
+    state: started
+
+- name: create postgres DB
+  postgresql_db:
+    name: dsm
+  become_user: postgres
+  become: yes
+
+- name: create postgres User
+  postgresql_user:
+    name: ansible
+    password: ansible
+    priv: "CONNECT,ALL"
+    db: dsm
+    role_attr_flags: SUPERUSER,CREATEDB,CREATEROLE,LOGIN
+  become_user: postgres
+  become: yes
+
+- name: Ensure local security method is changed to md5
+  ansible.builtin.lineinfile:
+    path: /var/lib/pgsql/11/data/pg_hba.conf
+    regexp: '^local*\s*\S+\s*\S+\s*peer'
+    state: absent
+
+- name: Ensure local security method is changed to md5
+  ansible.builtin.lineinfile:
+    path: /var/lib/pgsql/11/data/pg_hba.conf
+    regexp: '^host*\s*\S+\s*\S+\s*(?:[0-9]{1,3}\.){3}[0-9]{1,3}/[0-9]{1,2}'
+    state: absent
+
+- name: add new configuration to "pg_hba.conf"
+  blockinfile:
+    dest: /var/lib/pgsql/11/data/pg_hba.conf
+    block: |
+      local   all             all                                     md5
+      host    all             all             127.0.0.1/32            md5
+
+- name: Restart postgres service
```

Comment:
You should not need this step. Use instead `notify: restart postgresql` with the steps above.

Reply:
@goneri I was getting error: `ERROR! The requested handler 'restart postgresql-11' was not found in either the main handlers list nor in the listening handlers list`, so continued to use  the respective task instead.

Reply:
@goneri had tried that, but somehow it wasn't restarting the service as expected, coz I was getting following error with it:
```
"stderr_lines": [
        "Set Java locale to Locale.US - this occurs regardless of DSM locale.",
        "Running Upgrade Detector (Attempt: 1)",
        "No trace found. Using fresh install mode",
        "Upgrade Verification Screen settings accepted...",
        "Connecting to PostgreSQL...",
        "Unable to connect to database",
        "",
        "org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user \"ansible\"",
        "\tat org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:613)",
        "\tat org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:161)",
        "\tat org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:213)",
        "\tat org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:51)",
        "\tat org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:225)",
        "\tat org.postgresql.Driver.makeConnection(Driver.java:465)",
        "\tat org.postgresql.Driver.connect(Driver.java:264)",
        "\tat java.sql.DriverManager.getConnection(DriverManager.java:664)",
        "\tat java.sql.DriverManager.getConnection(DriverManager.java:208)",
        "\tat org.apache.commons.dbcp2.DriverManagerConnectionFactory.createConnection(DriverManagerConnectionFactory.java:123)",
        "\tat org.apache.commons.dbcp2.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:357)",
        "\tat org.apache.commons.pool2.impl.GenericObjectPool.create(GenericObjectPool.java:918)",
        "\tat org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:431)",
        "\tat org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:356)",
        "\tat org.apache.commons.dbcp2.PoolingDriver.connect(PoolingDriver.java:152)",
        "\tat java.sql.DriverManager.getConnection(DriverManager.java:664)",
        "\tat java.sql.DriverManager.getConnection(DriverManager.java:208)",
        "\tat com.thirdbrigade.persistence1.Persistence1Instance.getConnectionFromPool(Persistence1Instance.java:401)",
        "\tat com.thirdbrigade.persistence1.Persistence1Instance.getConnection(Persistence1Instance.java:360)",
        "\tat com.thirdbrigade.persistence1.ConnectionUtilities.begin(ConnectionUtilities.java:36)",
        "\tat com.trendmicro.ds.platform.objects.mt.Context.lambda$_beginT0Connection$2(Context.java:879)",
        "\tat com.trendmicro.ds.platform.objects.mt.DBSession.getConnection(DBSession.java:167)",
        "\tat com.thirdbrigade.persistence1.PersistentPeer.doExecuteDBValidationQuery(PersistentPeer.java:3017)",
        "\tat com.thirdbrigade.manager.core.install4j.screens.DatabaseScreen.validate(DatabaseScreen.java:737)",
        "\tat com.thirdbrigade.manager.core.install4j.screens.DatabaseScreen.handleUnattended(DatabaseScreen.java:502)",
        "\tat com.install4j.runtime.installer.frontend.headless.UnattendedScreenExecutor.handleScreen(UnattendedScreenExecutor.java:96)",
        "\tat com.install4j.runtime.installer.frontend.headless.AbstractHeadlessScreenExecutor.execute(AbstractHeadlessScreenExecutor.java:81)",
        "\tat com.install4j.runtime.installer.controller.Controller$1.run(Controller.java:138)",
        "\tat java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)",
        "\tat java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)",
        "\tat java.awt.EventQueue.access$500(EventQueue.java:97)",
        "\tat java.awt.EventQueue$3.run(EventQueue.java:709)",
        "\tat java.awt.EventQueue$3.run(EventQueue.java:703)",
        "\tat java.security.AccessController.doPrivileged(Native Method)",
        "\tat java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)",
        "\tat java.awt.EventQueue.dispatchEvent(EventQueue.java:728)",
        "\tat java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)",
        "\tat java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)",
        "\tat java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)",
        "\tat java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)",
        "\tat java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)",
        "\tat java.awt.EventDispatchThread.run(EventDispatchThread.java:82)",
        "\tSuppressed: org.postgresql.util.PSQLException: FATAL: Ident authentication failed for user \"ansible\"",
        "\t\tat org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:613)",
        "\t\tat org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:161)",
        "\t\tat org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:222)",
        "\t\t... 39 more"
    ],
    "stdout": "Unpacking JRE ...\nStarting Installer ...\nUnattended (Silent) mode started\nChecking for previous versions of Trend Micro Deep Security Manager...\nUpgrade Verification Screen settings accepted...\nThe installation directory has been set to /opt/dsm.\n[ERROR] Unable to connect to the database. An unexpected error has occurred. Please try again later. If the issue persists, contact your support provider.\nFATAL: Ident authentication failed for user \"ansible\"\n[ERROR] Database Screen settings rejected...\nRolling back changes ...",
    "stdout_lines": [
        "Unpacking JRE ...",
        "Starting Installer ...",
        "Unattended (Silent) mode started",
        "Checking for previous versions of Trend Micro Deep Security Manager...",
        "Upgrade Verification Screen settings accepted...",
        "The installation directory has been set to /opt/dsm.",
        "[ERROR] Unable to connect to the database. An unexpected error has occurred. Please try again later. If the issue persists, contact your support provider.",
        "FATAL: Ident authentication failed for user \"ansible\"",
        "[ERROR] Database Screen settings rejected...",
        "Rolling back changes ..."
    ]
```